src/SpendPermissionManager.sol

```diff
@@ -204,7 +292,35 @@ contract SpendPermissionManager is EIP712 {
     ///
     /// @return hash Hash of the spend permission.
     function getHash(SpendPermission memory spendPermission) public view returns (bytes32) {
-        return _hashTypedData(keccak256(abi.encode(MESSAGE_TYPEHASH, spendPermission)));
+        return _hashTypedData(keccak256(abi.encode(PERMISSION_TYPEHASH, spendPermission)));
+    }
+
+    /// @notice Hash a SpendPermissionBatch struct for signing in accordance with EIP-712.
+    ///
+    /// @param spendPermissionBatch Details of the spend permission batch.
+    ///
+    /// @return hash Hash of the spend permission batch.
+    function getBatchHash(SpendPermissionBatch memory spendPermissionBatch) public view returns (bytes32) {
+        uint256 tokenAllowancesLen = spendPermissionBatch.tokenAllowances.length;
+        bytes32[] memory tokenAllowanceHashes = new bytes32[](tokenAllowancesLen);
+        for (uint256 i; i < tokenAllowancesLen; i++) {
+            tokenAllowanceHashes[i] =
+                keccak256(abi.encode(TOKEN_ALLOWANCE_TYPEHASH, spendPermissionBatch.tokenAllowances[i]));
+        }
+
+        return _hashTypedData(
+            keccak256(
+                abi.encode(
+                    PERMISSION_BATCH_TYPEHASH,
+                    spendPermissionBatch.account,
+                    spendPermissionBatch.spender,
+                    spendPermissionBatch.start,
+                    spendPermissionBatch.end,
+                    spendPermissionBatch.period,
+                    keccak256(abi.encodePacked(tokenAllowanceHashes))
```

Comment:
the one addition that seemed missing from [uniswap's implementation](https://github.com/Uniswap/permit2/blob/main/src/libraries/PermitHash.sol) of a similar hash scheme was this `abi.encodePacked` around the array of hashes.

Reply:
Good catch!